book/tutorials/nb-to-package/intro.md

```diff
@@ -0,0 +1,270 @@
+You might ask yourself:
+
+- why would I want to package my code?
+- can't I just pass around notebooks?
+- what's the point of all this?
+- what is the meaning of life?
+
+... well, maybe not the last one. But the others are valid questions.
+
+## What is a Python package?
+
+A "package" or officially called a "distribution package" is simply
+a directory of Python modules and sub-packages with an `__init__.py` file
+that allows for the modular organization of code.
+It can be be installed and distributed for use by others.
+
+![Bedroom](https://images.unsplash.com/photo-1630699293259-0b6c08606c62?q=80&w=3870&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D)
+
+Let's think about this in a simpler term. We can think of a
+package like a room in a house. Say the "bedroom" is a package.
+Inside the bedroom, you have a bed, a nightstand, and a closet.
+
+- The bed and nightstand are like *modules* in a package.
+- The closet is like a *sub-package* in a package, since there might be other
+furniture pieces in a closet like a dresser; some closets can even be a whole room!
+
+### Module
+
+A module is a single file of Python code that can be imported into Python
+scripts or other modules,
+also known officially as an "import package".
+This is the smallest unit of code reuse in Python.
+You'll see these as `.py` files.
+
+As example, you can create a `hello_world` module by doing the following:
+
+1. Create a file called `hello_world.py` with the following content:
+
+    ```python
+    def say_hello():
+        print("Hello, World!")
+    ```
+
+2. Then, you can import this module in another Python script or notebook:
+
+    ```python
+    from hello_world import say_hello
+
+    say_hello()
+    ```
+
+    This will print `Hello, World!` to the console.
+
+This is a simple example of a module, but you can have more complex modules with many functions and classes.
+
+![Bed](https://images.unsplash.com/photo-1522771739844-6a9f6d5f14af?q=80&w=3871&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D)
+
+Going back to the bedroom analogy, a module is like a single piece of furniture in a room.
+For example, a bed is a piece of furniture that we spend most of our life on,
+this "bed" might be made up of various parts like a headboard, frame, and mattress.
+Each of these parts can be thought of as a function or class in a module.
+
+### Library
+
+A library is a collection of modules and packages that provide specific functionality or a set of related functionalities.
+In the Python ecosystem, a library is often used interchangeably with the term "package", and this can be confusing.
+For example, [Matplotlib](https://matplotlib.org/) is a plotting library that contains many modules to help you plot data
+or [Xarray](http://xarray.pydata.org/en/stable/) is a library that includes several modules and packages to help you work with labeled multi-dimensional arrays.
```

Comment:
Another example towards the beginning of this tutorial could identify the pieces from one of these larger libraries: 
```python
from xarray.testing import assert_equal
     ^      ^              ^
   library  module         function
```
https://docs.xarray.dev/en/stable/generated/xarray.testing.assert_equal.html